dev/system-test/pipeline.ts

```diff
@@ -0,0 +1,733 @@
+        Field.of('published').min().as('min_published')
+      )
+      .execute();
+    expectResults(results, {
+      count: 10,
+      max_rating: 4.7,
+      min_published: 1813,
+    });
+  });
+
+  it('can select fields', async () => {
+    const results = await randomCol
+      .pipeline()
+      .select('title', 'author')
+      .sort(Field.of('author').ascending())
+      .execute();
+    expectResults(
+      results,
+      {title: "The Hitchhiker's Guide to the Galaxy", author: 'Douglas Adams'},
+      {title: 'The Great Gatsby', author: 'F. Scott Fitzgerald'},
+      {title: 'Dune', author: 'Frank Herbert'},
+      {title: 'Crime and Punishment', author: 'Fyodor Dostoevsky'},
+      {
+        title: 'One Hundred Years of Solitude',
+        author: 'Gabriel García Márquez',
+      },
+      {title: '1984', author: 'George Orwell'},
+      {title: 'To Kill a Mockingbird', author: 'Harper Lee'},
+      {title: 'The Lord of the Rings', author: 'J.R.R. Tolkien'},
+      {title: 'Pride and Prejudice', author: 'Jane Austen'},
+      {title: "The Handmaid's Tale", author: 'Margaret Atwood'}
+    );
+  });
+
+  it('where with and', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(and(gt('rating', 4.5), eq('genre', 'Science Fiction')))
+      .execute();
+    expectResults(results, 'book10');
+  });
+
+  it('where with or', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(or(eq('genre', 'Romance'), eq('genre', 'Dystopian')))
+      .select('title')
+      .execute();
+    expectResults(
+      results,
+      {title: 'Pride and Prejudice'},
+      {title: "The Handmaid's Tale"},
+      {title: '1984'}
+    );
+  });
+
+  it('offset and limits', async () => {
+    const results = await randomCol
+      .pipeline()
+      .sort(Field.of('author').ascending())
+      .offset(5)
+      .limit(3)
+      .select('title', 'author')
+      .execute();
+    expectResults(
+      results,
+      {title: '1984', author: 'George Orwell'},
+      {title: 'To Kill a Mockingbird', author: 'Harper Lee'},
+      {title: 'The Lord of the Rings', author: 'J.R.R. Tolkien'}
+    );
+  });
+
+  it('arrayContains works', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(arrayContains('tags', 'comedy'))
+      .select('title')
+      .execute();
+    expectResults(results, {title: "The Hitchhiker's Guide to the Galaxy"});
+  });
+
+  it('arrayContainsAny works', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(arrayContainsAny('tags', ['comedy', 'classic']))
+      .select('title')
+      .execute();
+    expectResults(
+      results,
+      {title: "The Hitchhiker's Guide to the Galaxy"},
+      {title: 'Pride and Prejudice'}
+    );
+  });
+
+  it('arrayContainsAll works', async () => {
+    const results = await randomCol
+      .pipeline()
+      .where(Field.of('tags').arrayContainsAll('adventure', 'magic'))
```

Comment:
I wish this read `.where(field('tags').arrayContains...` or even `.where(Field.named('tags').arrayContains...`. As I read through the code, I continue to read it as "where field of something" but the meaning is "where field something".

Reply:
I think for Java Andy added an function overloads that covers the "common" usecase -- which is the case that the first parameter is a string specifying the name of the field. Is that possible to do for TS?